### PR TITLE
=per #17896 Support stashing of replayed in PersistentView

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #17896

**Comment by ktoso** [comment]
LGTM :)

File: akka-persistence/src/main/scala/akka/persistence/PersistentView.scala [hunk 1]
```diff
@@ -361,6 +361,10 @@ trait PersistentView extends Actor with Snapshotter with Stash with StashFactory
     override def recoveryRunning: Boolean = false
 
     override def stateReceive(receive: Receive, message: Any): Unit = message match {
+      case ReplayedMessage(p) ⇒
+        // we can get ReplayedMessage here if it was stashed by user during replay
+        // unwrap the payload
+        PersistentView.super.aroundReceive(receive, p.payload)
```
Comment (drewhk) [inline akka-persistence/src/main/scala/akka/persistence/PersistentView.scala]: hmm, is there a valid use case for stashing replayed messages?

Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/PersistentView.scala]: user asked for it, see #17896, and it is obviously not good to emit the internal `ReplayedMessage` wrapper

Comment (drewhk) [inline akka-persistence/src/main/scala/akka/persistence/PersistentView.scala]: yes, unwrapping is necessary if we support stashing, otherwise an explicit failure would be needed. It feels like a bit risky to stash replayed messages, but if users need it I trust your judgement to add support for it.

Comment (michaelom) [inline akka-persistence/src/main/scala/akka/persistence/PersistentView.scala]: >hmm, is there a valid use case for stashing replayed messages?

In our use case we are using a PersistentView to create a (denormalised) read model. Most of your implementations require no stashing - just mapping logic. However, sometimes we join replayed events with some data before writing to the read model. That data might not be available during event consumption (due to eventual consistency, load or a network failure). So we stash all incoming events for a few seconds, then unstash and check for availability of that data again. We repeat that until the data is available.

So basically we can delay event consumption with stash/unstash in a non-blocking fashion.

I am aware this could lead to OOMs but given the right circumstances (low event publishing frequency) it works very well.

Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/PersistentView.scala]: thanks for clarifying, the bug will be fixed in 2.4-M3
